dm/tests/shardddl4_1/run.sh

```diff
@@ -970,49 +970,72 @@ function DM_155_CASE {
 	sleep 1
 	random_restart 3
 
-	run_sql_source1 "insert into ${shardddl1}.${tb1} values(25,25,25,25,25);"
-	run_sql_source2 "insert into ${shardddl1}.${tb1} values(26,26,26,26,26);"
-	run_sql_source2 "insert into ${shardddl1}.${tb2} values(27,27,27);"
-
+	# make sure source1 has received conflict ddl and handled
+	check_log_contain_with_retry "skip conflict ddls in optimistic shard mode.*ALTER TABLE \`${shardddl1}\`.\`${tb1}\` CHANGE COLUMN \`c\` \`b\` INT" \
```

Comment:
Is it still possible that worker2/mysql2 is faster and here can't get this log, maybe we need to move check to [L913](https://github.com/pingcap/tiflow/pull/5868/files#diff-7b8de43fcae4ef1438831e3f9b2c6595c95f7377bdd3d89653a40b0099baae3eL913).

Reply:
`worker2/mysql2` has two shard tables, table2 will executed finish ddl in [L976](https://github.com/pingcap/tiflow/pull/5868/files#diff-7b8de43fcae4ef1438831e3f9b2c6595c95f7377bdd3d89653a40b0099baae3eL976) after we receive `worker1/mysql1`'s handle log here. So I think we won't have such problem.

Reply:
ping @WizardXiao